### PR TITLE
gh-251: Replaced all assertions with assertJ

### PR DESCRIPTION
**Pull request by t92549**

### CODE REVIEW

**Review by GCHQDeveloper314 — APPROVED** [review]
Certainly a good improvement to readability. Just a few very minor questions.

File: core/src/test/java/uk/gov/gchq/koryphe/impl/function/ToListTest.java [hunk 1]
```diff
@@ -43,7 +42,12 @@ public void shouldConvertNullToList() {
         final Object result = function.apply(null);
 
         // Then
-        assertEquals(Lists.newArrayList((Object) null), result);
+        assertThat(result)
+                .isEqualTo(Lists.newArrayList((Object) null))
+                .asList()
+                .isNotNull()
+                .isNotEmpty()
+                .containsExactly((Object) null);
```
[the remark this answers is not in the document]
Comment (t92549) [reply core/src/test/java/uk/gov/gchq/koryphe/impl/function/ToListTest.java:50]: So I agree that it isn't necessary to check that it is working correctly, but I felt that it was good to really verbosely clarify what was going on, as I thought it's kinda weird that when you convert null to a list, it makes a list with null inside it.

File: core/src/test/java/uk/gov/gchq/koryphe/impl/function/ToSetTest.java [hunk 2]
```diff
@@ -47,7 +46,12 @@ public void shouldConvertNullToHashSet() {
         final Object result = function.apply(null);
 
         // Then
-        assertEquals(Sets.newHashSet((Object) null), result);
+        assertThat(result)
+                .isEqualTo(Sets.newHashSet((Object) null))
+                .asInstanceOf(COLLECTION)
+                .isNotNull()
+                .isNotEmpty()
+                .containsExactly((Object) null);
```
[the remark this answers is not in the document]
Comment (t92549) [reply core/src/test/java/uk/gov/gchq/koryphe/impl/function/ToSetTest.java:54]: The same goes for my other reply too, I wanted to be really verbose that when you turn null into a `HashSet` that it isn't empty and actually contains null, but when you do it for a `TreeSet`, it returns an empty `TreeSet` and they cannot have null added to them.

File: core/src/test/java/uk/gov/gchq/koryphe/impl/predicate/AgeOffFromDaysTest.java [hunk 3]
```diff
@@ -40,10 +39,10 @@ public void shouldAcceptWhenWithinAgeOffLimit() {
         final AgeOffFromDays filter = new AgeOffFromDays();
 
         // When
-        final boolean accepted = filter.test(System.currentTimeMillis() - AGE_OFF_MILLISECONDS + MINUTE_IN_MILLISECONDS, AGE_OFF_DAYS);
+        Tuple2<Long, Integer> inputs = new Tuple2<Long, Integer>(System.currentTimeMillis() - AGE_OFF_MILLISECONDS + MINUTE_IN_MILLISECONDS, AGE_OFF_DAYS);
```
[the remark this answers is not in the document]
Comment (t92549) [reply core/src/test/java/uk/gov/gchq/koryphe/impl/predicate/AgeOffFromDaysTest.java:42]: Unfortunately due to how assertJ does it Predicate assertions it has to work like this, or be written as:
```java
assertThat(filter.test(System.currentTimeMillis() - AGE_OFF_MILLISECONDS + MINUTE_IN_MILLISECONDS, AGE_OFF_DAYS)).isTrue();
```
Alternatively, I could implement a custom assert class just for the AgeOffFromDays but it seemed overkill, overall, I thought being a bit verbose with the inputs is probably the nicest option and stays consistent.

[the remark this answers is not in the document]
Comment (GCHQDeveloper314) [reply core/src/test/java/uk/gov/gchq/koryphe/impl/predicate/AgeOffFromDaysTest.java:42]: I suspected that might be the case. Visually, using `filter.test` directly looks better, although as you say it is then inconsistent. Happy either way.

**Pull request merged**